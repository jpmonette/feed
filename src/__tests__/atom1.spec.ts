import { sampleFeed } from "./setup";

describe("atom 1.0", () => {
  it("should generate a valid feed", () => {
    let actual = sampleFeed.atom1();
    expect(actual).toMatchSnapshot();
  });

  it("should generate a valid feed - legacy", () => {
    let actual = sampleFeed.render("atom-1.0");
    expect(actual).toMatchSnapshot();
  });
});
