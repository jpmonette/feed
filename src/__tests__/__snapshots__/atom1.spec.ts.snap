// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atom 1.0 should generate a valid feed 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
<feed xmlns=\\"http://www.w3.org/2005/Atom\\">
    <id>http://example.com/</id>
    <title>Feed Title</title>
    <updated>2013-07-13T23:00:00.000Z</updated>
    <generator>https://github.com/jpmonette/feed</generator>
    <author>
        <name>John Doe</name>
        <email>johndoe@example.com</email>
        <uri>https://example.com/johndoe</uri>
    </author>
    <link rel=\\"alternate\\" href=\\"http://example.com/\\"/>
    <link rel=\\"self\\" href=\\"http://example.com/sampleFeed.rss\\"/>
    <link rel=\\"hub\\" href=\\"http://example.com/hub\\"/>
    <link rel=\\"next\\" href=\\"http://example.com/sampleFeed.rss?page=3\\"/>
    <link rel=\\"previous\\" href=\\"http://example.com/sampleFeed.rss?page=1\\"/>
    <link rel=\\"first\\" href=\\"http://example.com/sampleFeed.rss\\"/>
    <link rel=\\"last\\" href=\\"http://example.com/sampleFeed.rss?page=10\\"/>
    <subtitle>This is my personal feed!</subtitle>
    <logo>http://example.com/image.png</logo>
    <icon>http://example.com/feed.ico</icon>
    <rights>All rights reserved 2013, John Doe</rights>
    <category term=\\"Technology\\">
    </category>
    <contributor>
        <name>Johan Cruyff</name>
        <email>johancruyff@example.com</email>
        <uri>https://example.com/johancruyff</uri>
    </contributor>
    <entry>
        <title type=\\"html\\"><![CDATA[Hello World]]></title>
        <id>https://example.com/hello-world</id>
        <link href=\\"https://example.com/hello-world\\">
        </link>
        <updated>2013-07-13T23:00:00.000Z</updated>
        <summary type=\\"html\\"><![CDATA[This is an article about Hello World.]]></summary>
        <content type=\\"html\\"><![CDATA[Content of my item]]></content>
        <author>
            <name>Jane Doe</name>
            <email>janedoe@example.com</email>
            <uri>https://example.com/janedoe</uri>
        </author>
        <author>
            <name>Joe Smith</name>
            <email>joesmith@example.com</email>
            <uri>https://example.com/joesmith</uri>
        </author>
        <contributor>
            <name>Shawn Kemp</name>
            <email>shawnkemp@example.com</email>
            <uri>https://example.com/shawnkemp</uri>
        </contributor>
        <contributor>
            <name>Reggie Miller</name>
            <email>reggiemiller@example.com</email>
            <uri>https://example.com/reggiemiller</uri>
        </contributor>
        <published>2013-07-10T23:00:00.000Z</published>
        <rights>All rights reserved 2013, John Doe</rights>
    </entry>
</feed>"
`;
